sale_commission_advance/models/account_move.py

```diff
@@ -0,0 +1,129 @@
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def create(self, vals):
+        rec = super().create(vals)
+        if rec.has_related_sale_with_down_payment():
+            # to get correct commission amount
+            rec.recompute_lines_agents()
+        return rec
+
+    def has_related_sale_with_down_payment(self):
+        return any(
+            self.invoice_line_ids.mapped(
+                lambda x: x.sale_line_ids and x.sale_line_ids.is_downpayment
+            )
+        )
```

Comment:
This can be simplified this way:

```suggestion
        return any(self.invoice_line_ids.sale_line_ids.is_downpayment)
```

```
>>> any([])
False
```

Reply:
you will get  'bool' object is not iterable

Reply:
OK, I see, then it would be:

```suggestion
        dp_list = self.invoice_line_ids.sale_line_ids.is_downpayment
        return dp_list and any(dp_list)
```

Reply:
This is more readable, but also more efficient.